### PR TITLE
 Assignment-5 | GraphQL Api | Axios | Apollo Client Library  | HOC | Customized Component

### PR DESCRIPTION
**Pull request by kaushalyad321**

### CODE REVIEW

**Review by sumitbaudh22 — COMMENTED** [review]
image
Getting these console warnings, Please fix this

File: Assignment-5/app/src/components/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/index.js:1]: Please fix this file same as you have done in assignment 4

**Review by sumitbaudh22 — COMMENTED** [review]
fix code formatting in complete PR you can use ctrl+shft+i if you're using vs code

File: Assignment-5/app/src/components/FetchedData.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        setData(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error fetching data:", error);
+        setLoading(false);
+      });
+  }, []); 
+
+//   console.log(data)
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/FetchedData.js]: remove unnecessary comments

File: Assignment-5/app/src/components/ReapeatingDataFetching.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from "react";
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/ReapeatingDataFetching.js:1]: Change the filename to something meaningful like FetchDataWithErrorHandlers

File: Assignment-5/app/src/components/ReapeatingDataFetching.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+function FetchedData() {
+  const [data, setData] = useState([]);
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/ReapeatingDataFetching.js:4]: This task is not working properly if i click on refresh button it updates the data 2 two times on a single click

File: Assignment-5/app/src/components/DatafetchingSpinner.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from "react";
+
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/DatafetchingSpinner.js:2]: Spinner is not centered align

**Comment by kaushalyad321** [comment]
> fix code formatting in complete PR you can use ctrl+shft+i if you're using vs code

i have used in every pr

File: Assignment-5/app/src/components/index.js (hunk 1 above)
Comment (kaushalyad321) [reply Assignment-5/app/src/components/index.js:1]: > Please fix this file same as you have done in assignment
 updated

File: Assignment-5/app/src/components/ReapeatingDataFetching.js (hunk 3 above)
Comment (kaushalyad321) [reply Assignment-5/app/src/components/ReapeatingDataFetching.js:1]: ok

File: Assignment-5/app/src/components/ReapeatingDataFetching.js (hunk 4 above)
Comment (kaushalyad321) [reply Assignment-5/app/src/components/ReapeatingDataFetching.js:4]: fixing it

File: Assignment-5/app/src/components/PaginationFetchedData.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from "react";
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/PaginationFetchedData.js:1]: This task is not as per the acceptance, on pagination data should be fetch on click of next page

File: Assignment-5/app/src/components/DataFetchingAxios.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from "react";
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/DataFetchingAxios.js:1]: change file name to fetchDataUsingAxios

File: Assignment-5/app/src/components/DataFetchingAxios.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      setData(res.data);
+      setLoading(false) ;
+    })
+      .catch((error) => {
+        console.error("Error fetching data:", error);
+        setLoading(false);
+      });
+  }, []); 
+
+//   console.log(data)
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/DataFetchingAxios.js]: remove commented code

File: Assignment-5/app/src/components/DataFetchingAxios.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      <h1>Data fetched from API</h1>
+     {
+      Object.keys(data).map( (key)=> {
+              return <div key={key}>
+                 <span>{key}:</span>
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/DataFetchingAxios.js]: fix indentation

File: Assignment-5/app/src/components/ModifiedDataFetchingAxious.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from "react";
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/ModifiedDataFetchingAxious.js:1]: Correct the spelling of axios

File: Assignment-5/app/src/components/ModifiedDataFetchingAxious.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+import axios from 'axios'
+
+function ModifiedDataFetchingAxios() {
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+
```
Comment (sumitbaudh22) [inline Assignment-5/app/src/components/ModifiedDataFetchingAxious.js:7]: This task is not working properly if i click on refresh button it updates the data 2 two times on a single click

**Pull request merged**